subsys/usb/device_next/class/Kconfig.cdc_acm

```diff
@@ -21,6 +21,16 @@ config USBD_CDC_ACM_STACK_SIZE
 	help
 	  USB CDC ACM workqueue stack size.
 
+config USBD_CDC_ACM_TX_ZLP_ENABLED
+	bool "USB CDC ACM transmission always end with short packet"
```

Comment:
Arguably this could be potentially made `default y`. Windows driver is the original reason behind this, but Linux CDC ACM driver also can benefit from ZLP in corner case.

Linux CDC ACM driver will allocate URBs that can hold 2 Max Packet Size by default:
https://github.com/torvalds/linux/blob/c2ee9f594da826bea183ed14f2cc029c719bf4da/drivers/usb/class/cdc-acm.c#L1350-L1351C16
unless the `SINGLE_RX_URB` quirk is set.

If the CDC ACM device transmit data stream ends with packet that is exactly Max Packet Size bytes and ceases to send more data, on Linux host there is possibility that the last Max Packet Size packet will be withheld in (host side) URB until there is new data received from device. Depending on application protocol this may lead to stalls that are "fixed" with the `SINGLE_RX_URB` quirk.

The ZLP itself is completely invisible to higher level applications that use ttyACM interface (e.g. pyserial).

Reply:
Hi @tmon-nordic, Thanks for the information. I did not know about the `SINGLE_RX_URB` thing. It was set as default disabled as I was thinking that it is only meant for Windows Host Driver. Let's make this `default y` for now and see what the over reviewer thinks.  😄